### PR TITLE
feat: navigation tests - add page objects and tests (libero/editor#939)

### PR DESCRIPTION
**Pull request by will-byrne**

### CODE REVIEW

File: tests/navigation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import { MainText } from './page-objects/main-text';
+import { Navigation } from './page-objects/navigation';
+
+const setHeadingAndTestNavigation = async (page: Page, navigation: Navigation, heading: string): Promise<void> => {
+  const mainText = new MainText(page);
+  const text = 'To Ganymede and Titan';
+  await mainText.setText(text);
+  await mainText.selectText(text);
+  await mainText.setTextHeading(heading);
+  await navigation.clickLink(text);
```
Comment (NuclearRedeye) [inline tests/navigation.spec.ts:11]: So I wonder here if this is actually clicking the element that you expect, as you are using the same text. I can see below that you call this 4 times in the fixture 'dynamic main text headings', but the text will be the same each time. Hence won't the 2nd, 3rd and 4th calls actually click the first element?
Appending the heading to the `text` variable might solve this, or add an additional parameter to customise the text for each call.

File: tests/navigation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  await navigation.assertTargetInViewport(page.locator(`#mainText>div>div`).locator('text="To Ganymede and Titan"'));
```
Comment (NuclearRedeye) [inline tests/navigation.spec.ts:12]: Could use a template string here to avoid the repetition, e.g. `text="${text}"`

**Review by NuclearRedeye — CHANGES_REQUESTED** [review]
Overall everything looks good, but just think it's worth double checking that the `setHeadingAndTestNavigation` function is working as expected. If it is, then I'm happy to approve this.

File: tests/navigation.spec.ts (hunk 1 above)
Comment (will-byrne) [reply tests/navigation.spec.ts:11]: each test runs in a separate browser instance so these wont interfere with each other

File: tests/page-objects/navigation.ts [hunk 3]
```diff
@@ -16,9 +16,7 @@ export class Navigation {
         observer.disconnect();
       });
       observer.observe(element);
-      // Firefox doesn't call IntersectionObserver callback unless
-      // there are rafs.
-      requestAnimationFrame(() => {});
+      requestAnimationFrame(() => undefined);
```
Comment (NuclearRedeye) [inline tests/page-objects/navigation.ts:19]: I think it might be worth leaving that comment in, as it gives good context as to why that call is present.

**Review by NuclearRedeye — APPROVED** [review]

**Pull request merged**